### PR TITLE
zxing-cpp: add version 2.0.0, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zxing-cpp/***

- add version 2.0.0
- support conan v2
- add patch_description

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zxing-cpp/all/conanfile.py [hunk 1]
```diff
@@ -29,100 +31,89 @@ class ZXingCppConan(ConanFile):
...
-            cmake.definitions["BUILD_BLACKBOX_TESTS"] = False
+            tc.variables["BUILD_WRITERS"] = self.options.enable_encoders
+            tc.variables["BUILD_READERS"] = self.options.enable_decoders
+            tc.variables["BUILD_EXAMPLES"] = False
+            tc.variables["BUILD_BLACKBOX_TESTS"] = False
         if is_msvc(self):
-            cmake.definitions["LINK_CPP_STATICALLY"] = "MT" in str(self.settings.compiler.runtime)
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+            tc.variables["LINK_CPP_STATICALLY"] = "MT" in str(self.settings.compiler.runtime)
```
Comment (SpaceIm) [inline recipes/zxing-cpp/all/conanfile.py:96]:
```suggestion
            tc.variables["LINK_CPP_STATICALLY"] = is_msvc_static_runtime(self)
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/zxing-cpp/all/conandata.yml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "2.0.0":
+    url: "https://github.com/nu-book/zxing-cpp/archive/v1.4.0.tar.gz"
+    sha256: "126767bb56f8a1f25ae84d233db2e9b9be50d71f5776092d0e170ca0f0ed1862"
```
Comment (axxel) [inline recipes/zxing-cpp/all/conandata.yml:4]: That is the 1.4 url and hash, not the 2.0 one.

File: recipes/zxing-cpp/all/conanfile.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 import os
 
-required_conan_version = ">=1.43.0"
-
+required_conan_version = ">=1.53.0"
```
Comment (axxel) [inline recipes/zxing-cpp/all/conanfile.py:10]: a few lines below this one is the `homepage` field. It's contents is outdated. It should read `https://github.com/zxing-cpp/zxing-cpp`

File: recipes/zxing-cpp/all/conandata.yml (hunk 2 above)
Comment (prince-chrismc) [reply recipes/zxing-cpp/all/conandata.yml:4]: RIP, please open a new PR and tag me 🙏

**Pull request merged**